homeassistant/components/androidtv/media_player.py

```diff
@@ -133,35 +136,46 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
 
     if CONF_ADB_SERVER_IP not in config:
         # Use "adb_shell" (Python ADB implementation)
-        adb_log = "using Python ADB implementation " + (
-            f"with adbkey='{config[CONF_ADBKEY]}'"
-            if CONF_ADBKEY in config
-            else "without adbkey authentication"
-        )
-        if CONF_ADBKEY in config:
+        if CONF_ADBKEY not in config:
+            # Generate ADB key files (if they don't exist)
+            adbkey = hass.config.path(STORAGE_DIR, "androidtv_adbkey")
```

Comment:
If we would use a config entry and config flow we wouldn't need the custom storage code.

The config flow could also help with the user interaction to set up the integration.

Reply:
I looked into config flow, but 1) I've never worked with config flows before and I don't know when I would have time to learn it, code it up, and test it and 2) I like YAML config:)

How would using a config flow avoid the need for custom storage code?  Given that the config "data" that I want to store is a key pair and not typical JSON-serializable data, I thought that writing the files was the simplest approach.  

From Paulus on Discord (10/8 at 5:15 pm):

> You can just use the `.storage` folder.  helpers/storage.py helps with writing to that directory
> If your key is not JSON, feel free to just write to that dir
> try to map filename <> integration

Reply:
Sorry, I thought we could serialize the code.

The config flow could still help with the user interaction during integration set up.

Reply:
I agree that config flow would be helpful.  My hope is that someone more familiar with it could add that contribution!